base_geoengine/fields.py

```diff
@@ -0,0 +1,290 @@
+        """Create/update the column corresponding to ``self``.
+
+        For creation of geo column
+
+        :param model: an instance of the field's model
+        :param column: the column's configuration (dict)
+                       if it exists, or ``None``
+        """
+        # the column does not exist, create it
+
+        if not column:
+            create_geo_column(
+                model._cr,
+                model._table,
+                self.name,
+                self.geo_type,
+                self.srid,
+                self.dim,
+                self.string,
+            )
+            return
+
+        if column["udt_name"] == self.column_type[0]:
+            return
+
+        self.update_geo_db_column(model)
+
+        if column["udt_name"] in self.column_cast_from:
+            sql.convert_column(model._cr, model._table, self.name, self.column_type[1])
+        else:
+            newname = (self.name + "_moved{}").format
+            i = 0
+            while sql.column_exists(model._cr, model._table, newname(i)):
+                i += 1
+            if column["is_nullable"] == "NO":
+                sql.drop_not_null(model._cr, model._table, self.name)
+            sql.rename_column(model._cr, model._table, self.name, newname(i))
+            sql.create_column(
+                model._cr, model._table, self.name, self.column_type[1], self.string
+            )
+
+
+class GeoLine(GeoField):
+    """Field for POSTGIS geometry Line type"""
+
+    type = "geo_line"
+    geo_type = "LINESTRING"
+
+    @classmethod
+    def from_points(cls, cr, point1, point2, srid=None):
+        """
+        Converts given points in parameter to a line.
+        :param cr: DB cursor
+        :param point1: Point (BaseGeometry)
+        :param point2: Point (BaseGeometry)
+        :param srid: SRID
+        :return: LINESTRING Object
+        """
+        sql = """
+        SELECT
+            ST_MakeLine(
+                ST_GeomFromText(%(wkt1)s, %(srid)s),
+                ST_GeomFromText(%(wkt2)s, %(srid)s)
+            )
+        """
+        cr.execute(
+            sql,
+            {
+                "wkt1": point1.wkt,
+                "wkt2": point2.wkt,
+                "srid": srid or cls.srid,
+            },
+        )
+        res = cr.fetchone()
+        return cls.load_geo(res[0])
+
+
+class GeoPoint(GeoField):
+    """Field for POSTGIS geometry Point type"""
+
+    type = "geo_point"
+    geo_type = "POINT"
+
+    @classmethod
+    def from_latlon(cls, cr, latitude, longitude):
+        """Convert a (latitude, longitude) into an UTM coordinate Point:"""
+        pt = Point(longitude, latitude)
+        cr.execute(
+            """
+            SELECT
+                ST_Transform(
+                    ST_GeomFromText(%(wkt)s, 4326),
+                    %(srid)s)
+        """,
+            {"wkt": pt.wkt, "srid": cls.srid},
+        )
+        res = cr.fetchone()
+        return cls.load_geo(res[0])
+
```

Comment:
@kouffsamuel This commit is missing https://github.com/OCA/geospatial/commit/f5dfefb7d74a40c1c2684a019c7f1f609b6d8c26